src/core/algorithm/distributionToCred.js

```diff
@@ -10,25 +10,38 @@ import {type TimelineDistributions} from "./timelinePagerank";
 import {NodeAddress, type NodeAddressT} from "../../core/graph";
 
 /**
- * Represents the full timeline cred for a graph.
+ * Represents cred scores over time.
+ *
+ * It contains an array of intervals, which give timing information, and an
+ * array of CredTimeSlices, which are Float64Arrays. Each CredTimeSlice
+ * contains cred scores for an interval. The cred scores are included in
+ * node-address-sorted order, and as such the CredScores can only be
+ * interpreted in the context of an associated Graph.
+ *
+ * As invariants, it is guaranteed that:
+ * - intervals and intervalCredScores will always have the same length
+ * - all of the intervalCredScores will have a consistent implicit node ordering
+ *
+ * The type is marked opaque so that no-one else can construct instances that
+ * don't conform to these invariants.
  */
-export type FullTimelineCred = $ReadOnlyArray<{|
-  // The interval for this slice.
-  +interval: Interval,
-  // The cred for each node.
-  // (Uses the graph's canonical node ordering.)
-  +cred: Float64Array,
-|}>;
+export opaque type TimelineCredScores: {|
+  +intervals: $ReadOnlyArray<Interval>,
+  +intervalCredScores: $ReadOnlyArray<Float64Array>,
```

Comment:
Rather than explaining what the `$ReadOnlyArray<Float64Array>` is in the comments, I prefer to separate it to another alias.

```js
opaque type NodeOrderedCredScores = Float64Array;

{+intervalCredScores: $ReadOnlyArray<NodeOrderedCredScores>, ...}
```

_Note: would have nit-and-approved, but since this PR's purpose is to improve the API I'll add as comment._

Reply:
Would have preferred a nit-and-approve still so I can merge off that approval if I adopt your comment. 

I think adding a comment makes sense when asking a question, or when you would need to see the changed approach to know if you'd want to approve it. In this case, since you're proposing an explicit change outright, it feels sensible to just approve it.